src/widgets/FormDateInput.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import { Text, StyleSheet, TextInput } from 'react-native';
+import moment from 'moment';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import PropTypes from 'prop-types';
+
+import { FlexColumn } from './FlexColumn';
+import { FlexRow } from './FlexRow';
+
+import { APP_FONT_FAMILY, SUSSOL_ORANGE, DARKER_GREY, FINALISED_RED } from '../globalStyles';
+import { CalendarIcon } from './icons';
+import { CircleButton } from './CircleButton';
+
+export const FormDateInput = ({
+  value,
+  isRequired,
+  onValidate,
+  onChangeDate,
+  label,
+  invalidMessage,
+  invalidMessageStyle,
+  isRequiredStyle,
+  labelStyle,
+  textInputStyle,
+  placeholder,
+  placeholderTextColor,
+  underlineColorAndroid,
+}) => {
+  const initialValue = onValidate(value) ? new Date(value) : new Date();
+
+  const [inputState, setInputState] = React.useState({
+    isValid: true,
+    inputValue: moment(initialValue).format('DD/MM/YYYY'),
+    pickerSeedValue: initialValue,
+    datePickerOpen: false,
+  });
+
+  const { inputValue, isValid, pickerSeedValue, datePickerOpen } = inputState;
+
+  const IsRequiredLabel = () =>
+    (isRequired && <Text style={isRequiredStyle}>Is Required</Text>) || null;
+
+  const InvalidMessageLabel = () =>
+    !isValid && <Text style={invalidMessageStyle}>{invalidMessage}</Text>;
+
+  const onUpdate = (newValue, validity = true, pickerVisibility = false) => {
+    const newState = {
+      isValid: validity,
+      inputValue: newValue,
+      pickerSeedValue: validity ? moment(newValue, 'DD/MM/YYYY').toDate() : new Date(),
+      datePickerOpen: pickerVisibility,
+    };
+    setInputState(newState);
+    onChangeDate(newValue);
+  };
+
+  // When changing the value of the input, check the new validity and set the new input.
+  // Do not restrict input, but provide feedback to the user.
+  const onChangeTextCallback = React.useCallback(
+    newValue => {
+      const newValueIsValid = onValidate ? onValidate(newValue) : true;
+      onUpdate(newValue, newValueIsValid);
+    },
+    [onValidate]
+  );
+
+  const onChangeDatez = ({ nativeEvent }) => {
+    const { timestamp } = nativeEvent;
+    if (!timestamp) return;
+    const newDate = moment(new Date(timestamp)).format('DD/MM/YYYY');
+    onUpdate(newDate, true);
+  };
+
+  const openDatePicker = () => setInputState(state => ({ ...state, datePickerOpen: true }));
+
+  return (
+    <FlexRow flex={1}>
```

Comment:
Is there a reason for `{1}` and not `"1"`? isn't that ever so slightly more expensive to render than a plain string?

Reply:
I'm not sure what you mean - I'm guessing you're thinking `{1}` is a JS object? It's just JSX syntax for the parser. If we want a litteral object we need to `{{}}`.. Unless there's something else? Also `flex` needs a number, not a string